test/common/packagelib.py

```diff
@@ -37,8 +38,12 @@ def setUp(self):
             self.backend = "apt"
             self.primary_arch = "all"
             self.secondary_arch = "amd64"
+        elif re.match(r"fedora-(39|40)|(centos|rhel)-(8|9|10).*", self.machine.image):
+            self.backend = "dnf4"
```

Comment:
What about keeping `self.backend == "dnf"` and adding `self.backend_version == "5"`? Just for esthetics.

Reply:
I'd find that a bit confusing, I'd rather be explicit. I wonder what will happen first: dnf6 breaks everything again, images with dnf 4 go EOL :grin: